06_puppy/sirigithub/mapstore.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+)
+
+type MapStore map[int]Puppy
+
+func NewMapStore() MapStore {
+	return MapStore{}
+}
+
+// CreatePuppy takes a  user provided puppy, and creates a new Puppy in the store
+// returns the ID of the new Puppy.
+func (m MapStore) CreatePuppy(puppy *Puppy) int {
+	puppy.ID = len(m)
+	m[puppy.ID] = *puppy
+	return puppy.ID
+}
+
+// UpdatePuppy overrides an existing puppy in the store,
+// returns an error if id is not found or does not match the puppy ID
+func (m MapStore) UpdatePuppy(p *Puppy) error {
+	if _, exists := m[p.ID]; !exists {
```

Comment:
use `ok` instead of `exists` ([comma go idiom](https://golang.org/doc/effective_go.html#maps))

Reply:
below as well.

Reply:
fixed @juliaogris